### PR TITLE
bootstrap: Add recheck after bootstrap doDMLWork failed.

### PR DESCRIPTION
**Pull request by shenli**
Fix: https://github.com/pingcap/tidb/issues/1096
I will add test case latter in this PR.

### CODE REVIEW

**Comment by coocood** [comment]
LGTM

**Comment by shenli** [comment]
@ngaut PTAL

**Comment by shenli** [comment]
@zimulala PTAL

**Comment by c4pt0r** [comment]
LGTM

File: bootstrap.go [hunk 1]
```diff
@@ -206,7 +207,19 @@ func doDMLWorks(s Session) {
 		ON DUPLICATE KEY UPDATE VARIABLE_VALUE="%s"`,
 		mysql.SystemDB, mysql.TiDBTable, bootstrappedVar, bootstrappedVarTrue, bootstrappedVarTrue)
 	mustExecute(s, sql)
-	mustExecute(s, "COMMIT")
+	_, err := s.Execute("COMMIT")
+	if err != nil {
+		time.Sleep(1 * time.Second)
+		// Check if TiDB is already bootstraped.
```
Comment (siddontang) [inline bootstrap.go]: s/bootstraped/bootstrapped/

**Comment by shenli** [comment]
PTAL @siddontang

File: session_test.go [hunk 2]
```diff
@@ -1083,6 +1083,11 @@ func (s *testSessionSuite) TestBootstrap(c *C) {
 	mustExecSQL(c, se, "drop table if exists t")
 	se.Close()
 
+	// Try do bootstrap dml jobs on an already bootstraped TiDB system will not cause fatal.
+	// For https://github.com/pingcap/tidb/issues/1096
+	se, err = CreateSession(store)
```
Comment (ngaut) [inline session_test.go]: Why ignore error ?

Comment (shenli) [inline session_test.go]: Yes. And I should test on a new store.

**Comment by shenli** [comment]
PTAL @ngaut

**Comment by ngaut** [comment]
LGTM

**Pull request merged**